src/main/java/com/google/devtools/build/lib/skyframe/toolchains/SingleToolchainResolutionFunction.java

```diff
@@ -284,18 +293,41 @@ private static SingleToolchainResolutionValue resolveConstraints(
     return SingleToolchainResolutionValue.create(toolchainTypeInfo, resolvedToolchainLabels);
   }
 
+  /**
+   * Helper enum to define the three indentation levels used in {@link debugMessage}.
+   */
+  private static enum IndentLevel  {
+      TargetPlatformLevel(""),
+      ToolchainLevel("  "),
+      ExecutionPlatformLevel("    ");
+
+      final String value;
+
+      IndentLevel(String value) {
+          this.value = value;
+      }
+
+      @Override
+      public String toString() {
+          return value;
+      }
+  }
+
   /**
    * Helper method to print a debugging message, if the given {@link resolutionTrace} is not {@code
    * null}.
    */
   @FormatMethod
   private static void debugMessage(
-      @Nullable List<String> resolutionTrace, @FormatString String template, Object... args) {
+      @Nullable List<String> resolutionTrace,
+      IndentLevel indent,
+      @FormatString String template,
+      Object... args) {
     if (resolutionTrace == null) {
       return;
     }
-
-    resolutionTrace.add(String.format(template, args));
+    String padding = resolutionTrace.isEmpty() ? "" : " ".repeat("INFO: ".length());
```

Comment:
Why does this exist? To make the first line visible past the "INFO" header? I'm not sure it makes sense, please remove it.

Reply:
To align everything with the first INFO: line, as in

```
INFO: ToolchainResolution: Performing resolution of @bazel_tools//tools/cpp:toolchain_type for target platform @local_config_platform//:host
      ToolchainResolution:   Rejected toolchain @bazel_tools~cc_configure_extension~local_config_cc//:cc-compiler-armeabi-v7a; mismatching values: armv7, android
      ToolchainResolution:   Toolchain @bazel_tools~cc_configure_extension~local_config_cc//:cc-compiler-k8 is compatible with target plaform, searching for execution platforms:
```

I feel there is no point in having nice indents if you cannot see them, as would be the case with 

```
INFO: ToolchainResolution: Performing resolution of @bazel_tools//tools/cpp:toolchain_type for target platform @local_config_platform//:host
ToolchainResolution:   Rejected toolchain @bazel_tools~cc_configure_extension~local_config_cc//:cc-compiler-armeabi-v7a; mismatching values: armv7, android
ToolchainResolution:   Toolchain @bazel_tools~cc_configure_extension~local_config_cc//:cc-compiler-k8 is compatible with target plaform, searching for execution platforms:
```

Can you confirm you want this padding gone ?

Reply:
Leave it in: before this is merged I need to test with the internal version of bazel, which is using different output code.